operator/src/test/java/oracle/kubernetes/operator/steps/ServerDownIteratorStepTest.java

```diff
@@ -0,0 +1,383 @@
+  public void withConcurrencyOf1_bothClusteredServersShutdownSequentially() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(1).withReplicas(1);
+    addWlsCluster(CLUSTER, 8001, MS1, MS2);
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.FALSE));
+    testSupport.setTime(10, TimeUnit.SECONDS);
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withConcurrencyOf2_bothClusteredServersShutdownConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(2).withReplicas(1);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2);
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withConcurrencyOf0_clusteredServersShutdownConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(0);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2);
+
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withReplicaCountOf0AndConcurrencyOf1_clusteredServersShutdownConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(1).withReplicas(0);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2);
+
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withConcurrencyOf2AndReplicaCount1_3clusteredServersShutdownIn2Threads() {
```

Comment:
What is going on in this test? Why do the servers shut down in separate threads? There seems to be something important going on, and the name of the test doesn't clarify it. Is this actually two separate cases? If so, perhaps it should be two tests.

Reply:
The problem with the name is that it doesn't really inform the reader of what is going on. The point, apparently, is that the concurrency setting limits the number of simultaneous servers shutting down. The replica count is only relevant in that you have reduced it.

So maybe something like, `whenMaxConcurrentShutdownSet_limitNumberOfServersShuttingDownAtOnce`